feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_ubus.c

```diff
@@ -61,10 +61,18 @@ int ubus_set_channel_switch(const char *if_name, uint32_t frequency)
 
 	if (ubus_lookup_id(ubus, path, &id))
 		return -1;
-
 	blob_buf_init(&b, 0);
+
+	if (channel_bandwidth == 20 || 40) {
+		blobmsg_add_bool(&b, "ht", 1);
+	}
+	else if (channel_bandwidth == 80 || 160) {
+		blobmsg_add_bool(&b, "vht", 1);
+	}
```

Comment:
This is incorrect.  You need to pass in the htmode and select HT (11n), VHT (11ac) or HE (11a) appropriately.

The result of this code is that 20/40 run 802.11n and  80/169 run 802.11ac.  It will never do 802.11ax after a switch.  This will result in performance degradation similar to Ben's original report.

Reply:
In ubus channel switch api, i didn't see HE enable flag, and if I enable HT and VHT flags based on channel bandwidth, after channel switch somehow I saw it staying on HE mode on WiFi6 AP. I saw this behaviour in the Lab, when I was analyzing beacons using LANforge. I'll make sure this once again tomorrow in the LAB

Reply:
In ubus channel switch api, i didn't see HE enable flag, and if I enable HT and VHT flags based on channel bandwidth, after channel switch somehow I saw it staying on HE mode on WiFi6 AP. I saw this behaviour in the Lab, when I was analyzing beacons using LANforge. I'll make sure this once again tomorrow in the LAB